util/generate-sample-ledger/src/lib.rs

```diff
@@ -52,14 +57,28 @@ pub fn bootstrap_ledger(
     let mut previous_block: Option<Block> = None;
 
     let mut rng: FixedRng = SeedableRng::from_seed(seed.unwrap_or([33u8; 32]));
+    let mut hint_cursor = 0;
 
     for block_index in 0..num_blocks as u64 {
         println!("Creating block {} of {}.", block_index, num_blocks);
 
         let mut outputs: Vec<TxOut> = Vec::new();
         for recipient in recipients {
             for _i in 0..outputs_per_recipient_per_block {
-                outputs.push(create_output(recipient, picomob_per_output, &mut rng));
+                let hint_slice = if let Some(ht) = hint_text {
+                    let hint_chunk = ht.len() / num_outputs as usize;
```

Comment:
might be better to use `hint_chunk = min(ENCRYPTED_FOG_HINT_LEN, ht.len() - hint_cursor)` ?

Reply:
otherwise like, if you have a hint with 50 characters, and you have 1000 outputs, hint_chunk will be 0 every time due to rounding down, and no hints will be written at all

Reply:
@sugargoat it might be a good idea to make some kind of unit test for this?

Reply:
@garbageslam - I added an assert prior to making the outputs that the hint text divided among the txouts is > 0. We will need to decide whether we want all the hints to be the same in the origin block, or whether we want them all to be different, and likely padded. I think we might actually be leaning toward same.